app/helpers/patient_query_helper.rb

```diff
@@ -472,8 +472,15 @@ def advanced_filter_relative_date(patients, field, filter, tz_diff, type)
       timespan = filter[:value][:number].to_i.months if filter[:value][:unit] == 'months'
       return patients if timespan.nil?
 
-      timeframe = { after: (timespan.ago - tz_diff).beginning_of_day, before: local_current_time } if filter[:value][:when] == 'past'
-      timeframe = { after: local_current_time, before: (timespan.from_now - tz_diff).end_of_day } if filter[:value][:when] == 'next'
+      case filter[:value][:operator]
+      when 'less-than'
+        timeframe = { after: (timespan.ago - tz_diff).beginning_of_day, before: local_current_time } if filter[:value][:when] == 'past'
```

Comment:
I know you didn't make any changes to the logic of this part (`less-than`), but this is something I found when playing around with different timezones. I thought @kylietmo and I had figured this out a while ago but it seems like we might still be running into edge cases.

I set the browser time zone to hawaii time (5 hours behind) and this was done at 1AM EST or 8PM HST. I created a relative date filter for latest report within the past 10 days and here are the results, which according to our expected behavior, should include records with latest reports between the beginning of the day according to hawaiian time on 2/8 up to now

![Screen Shot 2021-02-19 at 1 00 42 AM](https://user-images.githubusercontent.com/9956561/108464454-29d87800-724e-11eb-8c14-f9540567e7ce.png)

It seems like, however, that not all 2/8 records are included since a similar query using past 11 days instead of past 10 days yields results from 2/8 that are not included in the results from the first query

![Screen Shot 2021-02-19 at 1 00 56 AM](https://user-images.githubusercontent.com/9956561/108464982-209bdb00-724f-11eb-9b3a-7b885a93908e.png)

I haven't been able to pinpoint the exact problem yet, and I'm not sure if it might have anything to do with the fact that the client and server are on different days or if it's something else, but when I set the browser back to EST, I'm still seeing the same problem where not all the results are properly being shown.

I'm also not sure if it could have something to do with the offset between the local dev server time zone and utc, here are snippets of the query in each time zone:

```
HST: (latest_assessment_at >= '2021-02-09 05:00:00') AND (latest_assessment_at <= '2021-02-19 05:48:32.805906')) subquery_for_count

PDT: (latest_assessment_at >= '2021-02-09 03:00:00') AND (latest_assessment_at <= '2021-02-19 05:50:14.667214')

EST: (latest_assessment_at >= '2021-02-09 00:00:00') AND (latest_assessment_at <= '2021-02-19 05:49:25.363425')
```

Reply:
I haven't fully looked into this, but date fields with a timestamp (this includes latest report) will be x units ago at the current time, so any monitorees with dates on that day but after that time wouldn't show up.  Could that be what you're seeing?

Reply:
OH are you talking about less than? I didn't change anything there so it would be an existing problem

Reply:
I haven't dug into this myself yet but @timwongj if you're sure there's still timezone issues let's make a bugticket and put it at the top of the backlog.

Reply:
https://tracker.codev.mitre.org/browse/SARAALERT-1343